### ISSUE TITLE
large number question

### ISSUE THREAD

**Issue by yinciki** [google-research/simclr#165]
hi there,

what is the LARGE_NUM in contrastive loss used for?

Thanks

**Comment by chentingpc** [comment]
offsetting/masking the identical pair similarity computation so that each augmented example compares with the other augmented view from the same original example, and augmented views of all other examples in the batch.

**Comment by yinciki** [comment]
> offsetting/masking the identical pair similarity computation so that each augmented example compares with the other augmented view from the same original example, and augmented views of all other examples in the batch.

Is it necessary or just for better performance?

**Comment by chentingpc** [comment]
We haven't ablated this choice empirically but it seems obvious to exclude
the cosine similarity computation of identical examples as the logits will
always be a constant 1.
